dask/dataframe/tests/test_shuffle.py

```diff
@@ -304,6 +304,30 @@ def test_set_index_3(shuffle_method):
     assert ddf2.npartitions == ddf.npartitions
 
 
+def test_set_index_no_sort():
+    """
+    GH10333 - Allow setting index on existing partitions without
+    computing new divisions and repartitioning.
+    """
+    df = pd.DataFrame({"col1": [2, 4, 1, 3, 5], "col2": [1, 2, 3, 4, 5]})
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    assert ddf.npartitions > 1
+    df_result = df.set_index("col1")
+
+    # Default is sort=True
+    # Index in ddf will be same values, but sorted
+    ddf_result = ddf.set_index("col1")
+    assert ddf_result.known_divisions
+    assert ddf_result.index.compute().tolist() == [1, 2, 3, 4, 5]
```

Comment:
```suggestion
    assert_eq(ddf_result, df_result.sort_index(), sort_results=False)
```
will test that the result matches the sorted version of the `pd.DataFrame`.